pymc3/smc/smc.py

```diff
@@ -462,23 +472,31 @@ def resample(self):
             self.chain_acc_rate = self.chain_acc_rate[self.resampling_indexes]
 
     def tune(self):
-        """Update proposal scales for each particle dimension"""
+        """Update proposal scales for each particle dimension and update number of MH steps"""
         if self.iteration > 1:
             # Rescale based on distance to 0.234 acceptance rate
             chain_scales = np.exp(np.log(self.proposal_scales) + (self.chain_acc_rate - 0.234))
             # Interpolate between individual and population scales
             self.proposal_scales = 0.5 * chain_scales + 0.5 * chain_scales.mean()
 
+            if self.tune_steps:
+                acc_rate = max(1.0 / self.proposed, self.chain_acc_rate.mean())
+                self.n_steps = min(
+                    self.max_steps,
+                    max(2, int(np.log(1 - self.p_acc_rate) / np.log(1 - acc_rate))),
+                )
+            self.proposed = self.draws * self.n_steps
+
     def mutate(self):
         """Metropolis-Hastings perturbation."""
         ac_ = np.empty((self.n_steps, self.draws))
 
         log_R = np.log(np.random.rand(self.n_steps, self.draws))
         for n_step in range(self.n_steps):
-            proposal = floatX(
-                self.tempered_posterior
-                + self.proposal_dist.rvs(size=self.draws) * self.proposal_scales[:, None]
-            )
+            proposal = self.proposal_dist.rvs(size=self.draws)
+            proposal = proposal.reshape(len(proposal), -1)
```

Comment:
Why is the reshape necessary?

Reply:
The test passed for IMH kernel and not for MH and when I check the dimensions were OK for IMH  but not MH. Also I find another issue with the MH kernel, but I will add it in a separated PR as is not related to the tuning of the steps and explains why in my simulations I was getting such poor results with the MH kernel.

Reply:
Y removed the reshaping I am not using the MultivariateProposal from pm.step_methods.metropolis as we used to do and because of that change I decided to also include in this PR the update of the covariance matrix and the multivariate proposal at each stage, that we used to do but we missed when we added the MH kernel back to SMC.